internal/handlers/commonparams/extract_params.go

```diff
@@ -37,10 +40,30 @@ import (
 //   - `opt` - field is optional;
 //   - `non-default` - field is unimplemented for non-default values;
 //   - `unimplemented` - field is not implemented yet;
-//   - `ignored` - field is ignored.
+//   - `ignored` - field is ignored;
+//   - `numericBool` - field is of type `bool` but can be parsed from numeric types;
+//   - `strict` - fields of numeric types are parsed strictly;
+//   - `positive` - fields of numeric types must be positive;
+//   - `numericAsBool` - fields of numeric types that must have as values only `0` or `1`.
 //
 // Collection field processed in a special way. For the commands that require collection name
 // it is extracted from the command name.
+// If the field could have different types (e.g. `*types.Document` and `*types.Array`) then
+// the field must be of type `any`.
```

Comment:
In that particular case it could (and probably should) be `types.CompositeTypeInterface`

Reply:
Some of the parameters we don't process right now, like `count.hint` could be either `string` or `*types.Document`. That's why I suggested using `any`.

Reply:
I think we should remove everything that is not being used: writing code "just in case" might we fine when design is frozen, but it is not frozen yet, and then we will have more code to update

Reply:
But if we remove everything that is not used we will return errors like 'field X is not expected' for the fields that could be used in some command. If we ok with that then I can remove all unused fields.

Reply:
Then it is being used, but indirectly… Let's wrap it up so there are no regressions (like we had with `count`). We will return to it later

Reply:
That being said, the original review comment still stands – that example is bad, the one in your response is much better. The comment should be updated.